admin: рассылки через админку — шаблоны, отложенные, Kafka

lost-znahid opened the pull request:
- CRUD шаблонов (Template), кампании (Campaign) в Django Admin
- немедленная отправка через публикацию события в Kafka
- отложенные/периодические задачи через django-apscheduler
- применены миграции admin/auth/sessions/contenttypes/broadcasts/apscheduler

lost-znahid committed:
admin: рассылки через админку — шаблоны, отложенные, Kafka

- CRUD шаблонов (Template), кампании (Campaign) в Django Admin
- немедленная отправка через публикацию события в Kafka
- отложенные/периодические задачи через django-apscheduler
- применены миграции admin/auth/sessions/contenttypes/broadcasts/apscheduler

lost-znahid committed:
Исправление кода для соответствия PEP8

lost-znahid committed:
feat(admin-service): Broadcasts MVP + планировщик и Kafka; lint/config правки

- Models: добавлены Template и Campaign (+ScheduleType, CampaignStatus), валидации и help_text.
- Admin: list_display/filters, readonly поля, кнопка “Запустить”, actions run_now/plan_selected, автопланирование при save, фикс P103.
- Scheduler: APScheduler с DjangoJobStore; поддержка DISABLE_SCHEDULER=1 для миграций; триггеры immediate/delayed/cron; ежедневная очистка; логирование.
- Kafka: ленивая инициализация KafkaProducer, JSON-сериализация, retries/linger, обработка ошибок.
- Config: загрузка .env, настройки messaging (KAFKA_BOOTSTRAP_SERVERS, KAFKA_TOPIC, SCHEDULER_TIMEZONE), чистка settings.
- Lint: правки E501/C812/I00x, удалены неиспользуемые импорты — flake8 проходит для admin_service.
- Migrations: добавлена broadcasts.0001_initial и миграции django_apscheduler.

lost-znahid committed:
feat(admin-service): Broadcasts MVP + планировщик и Kafka; lint/config правки

- Models: добавлены Template и Campaign (+ScheduleType, CampaignStatus), валидации и help_text.
- Admin: list_display/filters, readonly поля, кнопка “Запустить”, actions run_now/plan_selected, автопланирование при save, фикс P103.
- Scheduler: APScheduler с DjangoJobStore; поддержка DISABLE_SCHEDULER=1 для миграций; триггеры immediate/delayed/cron; ежедневная очистка; логирование.
- Kafka: ленивая инициализация KafkaProducer, JSON-сериализация, retries/linger, обработка ошибок.
- Config: загрузка .env, настройки messaging (KAFKA_BOOTSTRAP_SERVERS, KAFKA_TOPIC, SCHEDULER_TIMEZONE), чистка settings.
- Lint: правки E501/C812/I00x, удалены неиспользуемые импорты — flake8 проходит для admin_service.
- Migrations: добавлена broadcasts.0001_initial и миграции django_apscheduler.

lost-znahid committed:
Исправления по PEP8 в models.py и scheduler.py

lost-znahid committed:
Исправления по PEP8 в models.py

lost-znahid committed:
Исправления по PEP8 в models.py

lost-znahid committed:
Исправления конфликта форматов между black и flake8

lost-znahid committed:
Исправления ошибки линера 812 в файла scheduler.py и models.py

lost-znahid committed:
Лишняя запятая в scheduler.py

lost-znahid committed:
Лишняя запятая в scheduler.py

lost-znahid committed:
Неправильно установленная запятая в scheduler.py

lost-znahid committed:
снова конфликт между black и flake8. файлы models.py и scheduler.py

lost-znahid committed:
Исправление импортов в файлах apps.py и 0001_initial.pyи

mishakras commented on admin_service/broadcasts/services/users.py:
Я не вижу использования. Этот файл точно нужен по итогу?

lost-znahid replied:
Файл не нужен. Удалил

mishakras commented on admin_service/broadcasts/services/rendering.py:
Подозреваю, что этот файл не нужен, так как весь рендер должен идти в воркере.

lost-znahid replied:
Удалил

mishakras commented on admin_service/broadcasts/services/kafka_producer.py:
```diff
@@ -0,0 +1,102 @@
+class _KafkaWrapper:
+    """
+    Ленивая обёртка вокруг KafkaProducer, чтобы не падать на импорте при
+    старте Django (когда окружение Kafka ещё не готово).
```
Можно добавить цикл while который будет ждать пока kafka запускается, на всякий случай. Пример в модуле kafka_etl.

lost-znahid replied:
Реализовано

mishakras commented on admin_service/broadcasts/services/kafka_producer.py:
```diff
@@ -0,0 +1,102 @@
+            self._topic,
+            list(payload.keys()),
+        )
+        self._producer.send(self._topic, value=payload)
```
Стоит проверить и создать топик перед этим, при запуске.  Пример в том же модуле.

lost-znahid replied:
Реализовал

lost-znahid committed:
chore(broadcasts): правки по итогам ревью

- Удалены неиспользуемые файлы: services/users.py, services/rendering.py.
- KafkaProducer: добавлен цикл ожидания доступности брокеров (ретраи с backoff).
- KafkaProducer: проверка и автосоздание топика при старте до первой отправки.

lost-znahid committed:
Исправление кода под PEP8. Локально линеры ошибкок не давали

lost-znahid committed:
Две последние запятые пропустил
